### DISCUSSION TITLE
Batch Update for Outstatic

### DISCUSSION THREAD

**Discussion by HamadTheIronside** [avitorio/outstatic#251] [Ideas]
Hey Andre,

First of all, thank you for the amazing work on Outstatic! I recently discovered it, and I’ve been really enjoying it so far.

I’d like to suggest a feature that I think could add significant value to the project.

## Problem

Every action I take triggers Vercel to pull new changes, which can be inefficient and disruptive. This constant updating makes it harder to manage changes in bulk and can slow down the development process, especially that some metadata values become out of sync.

Also, one silly thing is seeing GitHub contributions getting spammed with tiny commits isn't fun 🚶🏻 

## Proposal

I believe introducing a batch update feature could address this issue. Instead of pushing changes immediately, it could temporarily store them (either in memory or log files) and allow the developer to push all updates together with a single click.

This would prevent unnecessary git pushes and make the workflow smoother.

Thanks for considering this suggestion! I’d be happy to provide more details or help further to develop if needed.

**Comment by avitorio** [comment]
Hey @HamadTheIronside,

Thank you for the suggestion and for sponsoring the project!

This is a great suggestion. I believe this would be solved by one of the following;

- local editing
- next.js incremental static regeneration
- branch switching and creation from the inteface

All of these are features I am working on. The branch switching feature will be launched on the next version. You'll be able to switch branches from the interface and then merge them when you want. I believe you can tell Vercel to ignore certain branches from automatic deployment.

I'll keep you posted. Cheers!

**Reply by HamadTheIronside** [reply]
Looking forward to the next versions 🚀 Currently, I am switching branches manually, and it does the work.
